ui/components/AutomationsTable.tsx

```diff
@@ -139,7 +147,7 @@ function AutomationsTable({ className, automations, hideSource }: Props) {
     <URLAddressableTable
       fields={fields}
       filters={initialFilterState}
-      rows={automations}
+      rows={sortAutomationsByDate(automations)}
```

Comment:
We (should) have sorting controls already on the `DataTable`. Can we set the default sort column to "Last Updated At" rather than sorting before passing them in.

@joshri Is there a way to select the "default" sort column?

Reply:
There is a `defaultSort` prop on DataTable which is looking for the index of the field you want to sort by. @ahussein3 does this actually fix 975? Won't other sorting cause the reordering issue? It's something that's been present since the polling. I thought previously that if we add a key (or maybe all objects already have a key?) we could use that as a backup sort value for cases where objects have the same value (which could be one reason why the sort isn't stable). If this fixes the reordering problems though that would be magical!

Reply:
If they `key` prop is the index of the the loop that is being iterated over, this will cause issues, since the order might be changing on every poll. Something to look into.

Reply:
Ooops, yes, @ahussein3 and I were talking about how to address this and I didn't really think it through.

I think we want to sort by the selected column _and then_ at least `(name, namespace, kind, clusterName)` which should make a unique set and provide a stable sort.

`_.sortBy(data, [selectedColumn, name, namespace, kind, clusterName])` or something?

Reply:
as part of this change could we make the default sort by Status so that failing/not ready items are highlighted?

Reply:
looks like sort by time is working as a key. I mean they all are sorted by time then by name. that's make it the rate of change is not that frequent like by name only. As data been called from different sources so they come in random shape so when sort by name it will get anyone first but as they are sorted by time so they are ordered and that item with that name will always be first. Sure this is not fixing the Issue but only reduce the rate of change. Hope I did explain that clearly. @joshri , @jpellizzari .
while we can set default sort I guess I can refactor that and go with the default sort instead.
As I can't see a final solution that can fix this issue. any suggestions ?